bin/generate_structs.jl

```diff
@@ -0,0 +1,136 @@
+
```

Comment:
I think this file goes in deps.

Reply:
This topic merits some discussion.  We could invoke this script in build.jl and then throw an error if any file has changed from git's perspective.  We need to enforce the rule that the generated files match the descriptor file.  Of course, we could remove the generated files from git and always run this script during build.  But you guys didn't like that.

Reply:
I don't get why these are mutually exclusive or why it should imply the removal of the generated code in git.

Reply:
1. Generated code is not in git.  The build script runs the generation script.  The descriptor file is always the source of truth.  Whenever the user downloads the repo they have to run build.  They will always get the correct version of generated code.  If the user changes the descriptor file but doesn't rerun the script, that is their fault.
2. Generated code is in git.  The build script runs the generation script and checks for changes.  We have to prevent the situation where a developer commits changes to the descriptor file but forgets to run the script.